### ISSUE TITLE
[COG-538] Identifying industry standard retrieval algorithms to implement

### ISSUE THREAD

**Issue by hajdul88** [topoteretes/cognee#225]
<sub>From [SyncLinear.com](https://synclinear.com) | [COG-538](https://linear.app/cognee/issue/COG-538/identifying-industry-standard-retrieval-algorithms-to-implement)</sub>

**Comment by hajdul88** [comment]
I will double check when implementing the retrievers. To include it everywhere would be a bigger task, will see what is the state and update beginning of next week.

**Comment by hajdul88** [comment]
I thought it was there, maybe it got dropped in some refactor. Please include it if you can. Our initial concept I did had it

**Comment by hajdul88** [comment]
Here I was thinking and reading research papers. I think we will need to have embeddings of each nodes as well as each edges. Most of the non LLM based retrievals rely on node and edge importances that are ultimately vector similarities to the query. As I know for now we have embeddings only for the documentchunks.

For this cycle I can do embeddings of these on the fly but its not gonna be production ready but will work for testing purposes, later these  will have to be moved to cognify pipeline and done when we are creating the graph. 

@lazar @Vasilije1990 what do you think?

**Comment by hajdul88** [comment]
Here I had a meeting with @lxobr where we discussed 2-3 different retrieval approaches. \[G-retriever, 2-step retriever, diffusion based retriever + community based filtering\] is the list I am aiming to develop this week). This list can change if the implementation of these would take too much time or we run into problems with any of these. With this I close this issue as we finished the review and collection of possible approaches for this cycle.
